core/ecschema-metadata/src/Metadata/Class.ts

```diff
@@ -76,10 +76,8 @@ export abstract class ECClass extends SchemaItem implements CustomAttributeConta
   public async getProperty(name: string, includeInherited: boolean = false): Promise<Property | undefined> {
     if (this.properties) {
       const upperKey = name.toUpperCase();
-      const foundProp = this.properties.find((prop) => prop.name.toUpperCase() === upperKey);
-      if (foundProp) {
-        return foundProp;
-      }
+      if (this.properties.has(upperKey))
+        return this.properties.get(upperKey);

```

Comment:
I think it would be better to get the property by key than check for undefined instead of checking to see if it exists by key then getting it by key.

Reply:
Made the change to get key first and then check for undefined